client/src/PublicRequest.ts

```diff
@@ -178,6 +185,73 @@ export type SignBtcTransactionRequest
     = SignBtcTransactionRequestStandard
     | SignBtcTransactionRequestCheckout;
 
+export type SignSwapRequest = SimpleRequest & {
+    swapId: string,
+    fund:
+        {type: 'NIM'}
+            & Omit<TransactionInfo,
+                'senderType'
+                | 'recipient'
+                | 'recipientType'
+                | 'recipientLabel'
+                | 'validityStartHeight'
+                | 'data'
+                | 'flags'
+            >
+            & { senderLabel: string }
+        | {type: 'BTC', keyPaths: string[], value: number, fee: number },
+    redeem:
+        {type: 'NIM'}
+            & Omit<TransactionInfo,
+                'sender'
+                | 'senderType'
+                | 'senderLabel'
+                | 'recipientType'
+                | 'validityStartHeight'
+                | 'data'
+                | 'flags'
+            >
+            & { recipientLabel: string }
+        | {type: 'BTC', keyPaths: string[], value: number, fee: number },
+
+    // Data needed for display
+    fiatCurrency: string,
+    nimFiatRate: number,
+    btcFiatRate: number,
+    serviceFundingNetworkFee: number, // Luna or Sats, depending which one gets funded
+    serviceRedeemingNetworkFee: number, // Luna or Sats, depending which one gets redeemed
+    serviceExchangeFee: number, // Luna or Sats, depending which one gets funded
+    nimiqAddresses: Array<{
+        address: string,
+        balance: number, // Luna
+    }>,
+    bitcoinAccount: {
+        balance: number, // Sats
+    },
+};
+
+// Used in swap-iframe
+export type SignSwapTransactionsRequest = {
+    swapId: string,
+    fund:
+        {type: 'NIM'}
+            & Omit<TransactionInfo, 'keyPath' | 'recipient' | 'recipientType' | 'data'>
```

Comment:
- `keyPath` was already provided and the private key derived in the top-level request above
- `recipient` is generated automatically for HTLC creation transactions
- `recipientType` is clear, must be `HTLC`
- `data` is omitted because it's optional in `TransactionInfo`, but I am re-adding it below without the optionality.